extensions/arc/deployment/src/main/java/io/quarkus/arc/deployment/ConfigBuildStep.java

```diff
@@ -250,40 +273,77 @@ void generateConfigClasses(
     }
 
     @BuildStep
-    void beanConfigClasses(
+    void registerConfigMappingsBean(
+            BeanRegistrationPhaseBuildItem beanRegistration,
             List<ConfigClassBuildItem> configClasses,
-            BeanRegistrationPhaseBuildItem beanRegistrationPhase,
             CombinedIndexBuildItem combinedIndex,
-            BuildProducer<BeanConfiguratorBuildItem> beanConfigurationRegistry) {
+            BuildProducer<BeanConfiguratorBuildItem> beanConfigurator) {
+
+        Set<ConfigClassBuildItem> configMappings = new HashSet<>();
+        for (InjectionPointInfo injectionPoint : beanRegistration.getInjectionPoints()) {
+            Type type = injectionPoint.getType();
+            for (ConfigClassBuildItem configClass : configClasses) {
+                if (configClass.isMapping()
+                        && type.name().equals(DotName.createSimple(configClass.getConfigClass().getName()))) {
+                    configMappings.add(configClass);
+                    break;
+                }
+            }
+        }
 
-        for (ConfigClassBuildItem configClass : configClasses) {
-            if (configClass.getGeneratedClasses().isEmpty()) {
+        for (ConfigClassBuildItem configClass : configMappings) {
+            beanConfigurator.produce(new BeanConfiguratorBuildItem(
+                    beanRegistration.getContext()
+                            .configure(configClass.getConfigClass())
```

Comment:
You should also make it unremovable via `configurator.unremovable()` if the config interface is annotated with `@Unremovable` because ArC ignores annotations when registering a synthetic bean...

Reply:
Sure.